datafusion/optimizer/tests/integration-test.rs

```diff
@@ -29,6 +29,33 @@ use std::any::Any;
 use std::collections::HashMap;
 use std::sync::Arc;
 
+#[test]
+fn case_when() -> Result<()> {
+    let sql = "SELECT CASE WHEN col_int32 > 0 THEN 1 ELSE 0 END FROM test";
+    let plan = test_sql(sql)?;
+    let expected = "Projection: CASE WHEN #test.col_int32 > Int32(0) THEN Int64(1) ELSE Int64(0) END\
```

Comment:
This plan doesn't look ideal to me -- since `col_int32` is int32 I think the exprs could be rewritten to 


```
CASE WHEN #test.col_int32 > Int32(0) THEN Int32(1) ELSE Int32(0) END
```

Could be done as a follow on PR for sure

Reply:
https://github.com/apache/arrow-datafusion/issues/3702